linera-core/src/client.rs

```diff
@@ -1307,12 +1305,17 @@ where
         };
         // Collect the hashed certificate values required for execution.
         let committee = self.local_committee().await?;
-        let nodes: Vec<(ValidatorName, P::Node)> =
-            self.validator_node_provider.make_nodes(&committee)?;
+        let nodes = self
+            .validator_node_provider
+            .make_nodes::<Vec<_>>(&committee)?;
         let values = self
             .client
             .local_node
-            .read_or_download_hashed_certificate_values(nodes.clone(), block.bytecode_locations())
+            .clone()
+            .read_or_download_hashed_certificate_values(
+                nodes.clone(),
+                block.bytecode_locations().into_keys(),
```

Comment:
If we don't need the sender chain ID anymore, should we return only the locations from `Block::bytecode_locations`?

Reply:
Good point!